src/ansi-c/c_expr.h

```diff
@@ -200,4 +200,28 @@ inline side_effect_expr_overflowt &to_side_effect_expr_overflow(exprt &expr)
   return static_cast<side_effect_expr_overflowt &>(side_effect_expr);
 }
 
+/// \brief A base class for an expression that indicates the pre-function-call
+/// value of an expression passed as a parameter to a function
+class old_exprt : public unary_exprt
+{
+public:
+  explicit old_exprt(irep_idt id, exprt variable)
+    : unary_exprt(id, std::move(variable))
+  {
+  }
+
+  const exprt &parameter() const
+  {
```

Comment:
"Parameter" is wrong terminology here. Given that you don't support symbols yet, the expression that's returned is actually never a parameter. Call it "expression".